wgpu-hal/src/metal/device.rs

```diff
@@ -1272,3 +1312,22 @@ impl crate::Device for super::Device {
         unimplemented!()
     }
 }
+
+// TODO: `label` is not exposed on `MTLBuffer`, is this always correct?
+fn set_buffer_label(buffer: &ProtocolObject<dyn MTLBuffer>, label: Option<&NSString>) {
+    unsafe { msg_send![buffer, setLabel: label] }
+}
+
+// TODO: `label` is not exposed on `MTLTexture`, is this always correct?
+fn set_texture_label(buffer: &ProtocolObject<dyn MTLTexture>, label: Option<&NSString>) {
+    unsafe { msg_send![buffer, setLabel: label] }
+}
+
+// TODO: `newComputePipelineStateWithDescriptor:error:` is not exposed on
+// `MTLDevice`, is this always correct?
+fn new_compute_pipeline_state_with_descriptor(
+    device: &ProtocolObject<dyn MTLDevice>,
+    descriptor: &MTLComputePipelineDescriptor,
+) -> Result<Retained<ProtocolObject<dyn MTLComputePipelineState>>, Retained<NSError>> {
+    unsafe { msg_send_id![device, newComputePipelineStateWithDescriptor: descriptor, error: _] }
+}
```

Comment:
It's odd that it doesn't exist since `newRenderPipelineStateWithDescriptor:error:` does.
We could use `newComputePipelineStateWithDescriptor:options:reflection:error:` though and pass `MTLPipelineOptionNone` for `options` and `nil` for `reflection`.